### PR TITLE
Decrease size of product-tests-launcher jar

### PR DESCRIPTION
**Pull request by losipiuk**
Decrease size of product-tests-launcher jar## Release notes

(s) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

File: testing/trino-product-tests-launcher/pom.xml [hunk 1]
```diff
@@ -21,6 +21,13 @@
         <dependency>
             <groupId>io.trino</groupId>
             <artifactId>trino-plugin-reader</artifactId>
+            <exclusions>
+                <exclusion>
+                    <!-- trino main pulls a lot of dependencies pumping jar size, and we only need categories definitions defined in PluginReader class -->
+                    <groupId>io.trino</groupId>
+                    <artifactId>trino-main</artifactId>
```
Comment (findepi) [inline testing/trino-product-tests-launcher/pom.xml:28]: Also, by their design, prroduct tests should be "black box". They intentionally never depended on the trino-main.
Can you perhaps enhance the comment to indicate that too?

Also, why does the plugin reader pull trino-main?
Is it a fake dep? Does plugin reader combine different functionalities, some depending on trino-main and some not? maybe plugin reader should be two separate modules?

**Review by findepi — APPROVED** [review]

File: testing/trino-product-tests-launcher/pom.xml (hunk 1 above)
Comment (nineinchnick) [reply testing/trino-product-tests-launcher/pom.xml:28]: PluginReader uses this single function from `trino-main` - `extractFunctions`: https://github.com/trinodb/trino/blob/master/testing/trino-plugin-reader/src/main/java/io/trino/server/PluginReader.java#L136

The product tests launcher only needs a few constants from the plugin reader. It definitely feels like something should be moved to improve dependencies, but I'm not sure what.

Comment (findepi) [reply testing/trino-product-tests-launcher/pom.xml:28]: > The product tests launcher only needs a few constants from the plugin reader. I

Can we move these constants out to `trino-plugin-reader-{api,spi,something}` jar?

Comment (losipiuk) [reply testing/trino-product-tests-launcher/pom.xml]: Yeah - it crossed my mind too. I was not convinced that we want extra module just for that. Probably the cleanest approach yet.
I will merge this one as a stop-gap though.

**Pull request merged**